go/arrow/array/float16.go

```diff
@@ -21,13 +21,13 @@ import (
 	"strings"
 
 	"github.com/apache/arrow/go/arrow"
-	"github.com/apache/arrow/go/arrow/numeric"
+	"github.com/apache/arrow/go/arrow/float16"
 )
 
 // A type which represents an immutable sequence of Float16 values.
 type Float16 struct {
 	array
-	values []numeric.Float16
+	values []float16.Float16
```

Comment:
I would like to remove the stutter if possible. `float16.Value` perhaps? @sbinet thoughts?

Reply:
SGTM.
FTR, here was my previous laundry list: https://github.com/apache/arrow/pull/4083#discussion_r273410118

Reply:
As an alternative to `float16.Value`, perhaps `float16.Num` or `float16.Number`?

I apologize, as this does feel like we're bike shedding.

Reply:
`float16.Num` SGTM.